exec/java-exec/src/test/java/org/apache/drill/exec/record/vector/TestDictVector.java

```diff
@@ -0,0 +1,460 @@
+          mapReader.next();
+          Float actualKey = mapReader.reader(DictVector.FIELD_KEY_NAME).readFloat();
+          Long actualValue = mapReader.reader(DictVector.FIELD_VALUE_NAME).readLong();
+          Assert.assertEquals(entry.getKey(), actualKey);
+          Assert.assertEquals(entry.getValue(), actualValue);
+        }
+      }
+
+      newMapVector.clear();
+      byteBuf.release();
+
+      writableBatch.clear();
+    }
+  }
+
+  @Test
+  public void testLoadBatchLoader() throws Exception {
+    MaterializedField field = MaterializedField.create("map", DictVector.TYPE);
+    try (DictVector mapVector = new DictVector(field, allocator, null)) {
+      mapVector.allocateNew();
+
+      List<Map<Object, Object>> maps = Arrays.asList(
+          TestBuilder.mapOfObject(4f, 1L, 5.3f, 2L, 0.3f, 3L, -0.2f, 4L, 102.07f, 5L),
+          TestBuilder.mapOfObject(45f, 6L, 9.2f, 7L),
+          TestBuilder.mapOfObject(4.01f, 8L, 9.2f, 9L, -2.3f, 10L),
+          TestBuilder.mapOfObject(),
+          TestBuilder.mapOfObject(11f, 11L, 9.73f, 12L, 0.03f, 13L)
+      );
+
+      BaseWriter.DictWriter mapWriter = new SingleDictWriter(mapVector, null);
+      int index = 0;
+      for (Map<Object, Object> map : maps) {
+        mapWriter.setPosition(index++);
+        mapWriter.start();
+        for (Map.Entry<Object, Object> entry : map.entrySet()) {
+          mapWriter.startKeyValuePair();
+          mapWriter.float4(DictVector.FIELD_KEY_NAME).writeFloat4((float) entry.getKey());
+          mapWriter.bigInt(DictVector.FIELD_VALUE_NAME).writeBigInt((long) entry.getValue());
+          mapWriter.endKeyValuePair();
+        }
+        mapWriter.end();
+      }
+
+      WritableBatch writableBatch = WritableBatch.getBatchNoHV(maps.size(), Collections.singletonList(mapVector), false);
+      // Serialize the vector
+      DrillBuf byteBuf = TestLoad.serializeBatch(allocator, writableBatch);
+      RecordBatchLoader batchLoader = new RecordBatchLoader(allocator);
+      batchLoader.load(writableBatch.getDef(), byteBuf);
+
+      byteBuf.release();
+
+      assertEquals(maps.size(), batchLoader.getRecordCount());
+
+      writableBatch.clear();
+      batchLoader.clear();
+    }
+  }
+
+  @Test
+  public void testGetByKey() {
+    MaterializedField field = MaterializedField.create("map", DictVector.TYPE);
+    try (DictVector mapVector = new DictVector(field, allocator, null)) {
+      mapVector.allocateNew();
+
+      List<Map<Object, Object>> maps = Arrays.asList(
+          TestBuilder.mapOfObject(4f, 1L, 5.3f, 2L, 0.3f, 3L, -0.2f, 4L, 102.07f, 5L),
+          TestBuilder.mapOfObject(45f, 6L, 9.2f, 7L),
+          TestBuilder.mapOfObject(4.01f, 8L, 4f, 9L, -2.3f, 10L),
+          TestBuilder.mapOfObject(-2.5f, 11L),
+          TestBuilder.mapOfObject(),
+          TestBuilder.mapOfObject(11f, 12L, 9.73f, 13L, 4f, 14L)
+      );
+
+      BaseWriter.DictWriter mapWriter = new SingleDictWriter(mapVector, null);
```

Comment:
This code : 
```java
      BaseWriter.DictWriter mapWriter = new SingleDictWriter(mapVector, null);
      int index = 0;
      for (Map<Object, Object> map : maps) {
        mapWriter.setPosition(index++);
        mapWriter.start();
        for (Map.Entry<Object, Object> entry : map.entrySet()) {
          mapWriter.startKeyValuePair();
          mapWriter.float4(DictVector.FIELD_KEY_NAME).writeFloat4((float) entry.getKey());
          mapWriter.bigInt(DictVector.FIELD_VALUE_NAME).writeBigInt((long) entry.getValue());
          mapWriter.endKeyValuePair();
        }
        mapWriter.end();
      }
``` 
is repeated few times in the class. Please extract it to helper method named like ```writeMapsToVector```.